src/main/fc/fc_msp.c

```diff
@@ -2154,10 +2169,19 @@ static mspResult_e mspCommonProcessInCommand(uint8_t cmdMSP, sbuf_t *src)
 #endif
 #if defined(OSD)
                 osdConfigMutable()->units = sbufReadU8(src);
+
+                // Alarms
                 osdConfigMutable()->rssi_alarm = sbufReadU8(src);
                 osdConfigMutable()->cap_alarm = sbufReadU16(src);
-                sbufReadU16(src);
+                sbufReadU16(src); // Skip unused (previously fly timer)
                 osdConfigMutable()->alt_alarm = sbufReadU16(src);
+
+                // Timers
+                for (int i = 0; i < OSD_TIMER_COUNT; i++) {
```

Comment:
Yeah, legacy code isn't doing it in most places, and it typically creates a headache with respect to not breaking compatibility with legacy apps when adding new stuff.